pkg/doc/verifiable/credential.go

```diff
@@ -587,56 +608,50 @@ func NewCredential(vcData []byte, opts ...CredentialOpt) (*Credential, []byte, e
 		return nil, nil, fmt.Errorf("unmarshal new credential: %w", err)
 	}
 
-	// Load custom credential schemas if defined.
-	var schemas []TypedID
-	if raw.Schema != nil {
-		schemas, err = loadCredentialSchemas(vcDataDecoded)
-		if err != nil {
-			return nil, nil, fmt.Errorf("load schemas of new credential: %w", err)
-		}
-	} else {
-		schemas = make([]TypedID, 0)
-	}
-
-	// Validate raw credential.
-	err = validate(vcDataDecoded, schemas, vcOpts)
-	if err != nil {
-		return nil, nil, fmt.Errorf("validate new credential: %w", err)
-	}
-
 	// Create credential from raw.
-	vc, err := newCredential(&raw, schemas)
+	vc, err := newCredential(&raw)
 	if err != nil {
 		return nil, nil, fmt.Errorf("build new credential: %w", err)
 	}
 
-	err = postValidateCredential(vc, vcOpts)
+	err = validateCredential(vc, vcDataDecoded, vcOpts)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return vc, vcDataDecoded, nil
 }
 
-func postValidateCredential(vc *Credential, vcOpts *credentialOpts) error {
+func validateCredential(vc *Credential, vcBytes []byte, vcOpts *credentialOpts) error {
 	// Credential and type constraint.
 	switch vcOpts.modelValidationMode {
-	case jsonldValidation:
-		// todo support JSON-LD validation (https://github.com/hyperledger/aries-framework-go/issues/952)
+	case adaptedValidation:
+		err := vc.validateJSONSchema(vcBytes, vcOpts)
```

Comment:
Why do we also need to validateJSONSchema when len(vc.Context) > 1?

Reply:
Agree! I added a comment here and also renamed the validation mode to `combinedValidation`. Also, a more detailed explanation of `combinedValidation` mode is provided at the definition.